frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx

```diff
@@ -1,23 +1,52 @@
 import * as React from 'react';
 import { DetailsPage, DetailsPageProps } from '@console/internal/components/factory';
-import { navFactory } from '@console/internal/components/utils';
+import { Kebab, navFactory } from '@console/internal/components/utils';
+import { k8sGet } from '@console/internal/module/k8s';
+import { ErrorPage404 } from '@console/internal/components/error';
+import { PipelineRunModel } from '../../models';
+import {
+  rerunPipeline,
+  stopPipelineRun,
+  handlePipelineRunSubmit,
+} from '../../utils/pipeline-actions';
 import { PipelineRunDetails } from './PipelineRunDetails';
 import { PipelineRunLogsWithActiveTask } from './PipelineRunLogs';
 
-const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => (
-  <DetailsPage
-    {...props}
-    pages={[
-      navFactory.details(PipelineRunDetails),
-      navFactory.editYaml(),
-      {
-        href: 'logs',
-        path: 'logs/:name?',
-        name: 'Logs',
-        component: PipelineRunLogsWithActiveTask,
-      },
-    ]}
-  />
-);
+const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => {
+  const [plrInstance, setPlrInstance] = React.useState({ ref: null, status: null });
 
+  React.useEffect(() => {
+    k8sGet(PipelineRunModel, props.name, props.namespace)
```

Comment:
Do not fetch the pipeline run resource here. The details page will do that for you and return the instance in the action callback.
The details page also does a watch so that it is updated in real time.